lib/Recur/RRuleIterator.php

```diff
@@ -588,6 +588,11 @@ protected function nextYearly() {
                     (int)$currentDayOfMonth
                 );
 
+                // maximum year 10000 should be safe to stop (2016 + max 3500 yearly occurrences)
+                if ((int)$currentYear > 10000) {
+                    throw new NoInstancesException('This recurrence rule does not generate any valid instances, stopped in year 10000');
```

Comment:
Good point! Maybe we do need an exception but then catch it elsewhere. The goal ultimately is that the `valid()` function needs to return false for these cases.